### PR TITLE
Analytics: Displays analytics settings for sites with Premium plans and above

### LINKED ISSUE #35737: GA: Make GA available to Premium plans
Work already advanced by Robert here: https://github.com/Automattic/wp-calypso/pull/32478

Need to be adjusted to Premium plans and not all paid plans as built by Robert so far. 

Listing bits and pieces, that need updating:

- [ ] The support docs - https://en.support.wordpress.com/google-analytics/
- [ ] The plan features - https://wordpress.com/pricing/
- [ ] Update the API to ensure the plugin is enabled - It looks like the setting is saved for all sites, but the plugin is only enabled on Business plans

### PR DESCRIPTION
**Pull request by creativecoder**
#### Changes proposed in this Pull Request

Adds Google Analytics to WordPress.com sites with Premium plans. (Sites with Business plans continue to have the feature, as well.)

Show the Analytics settings (Tools > Marketing > Traffic) for sites with Premium plans
Screen Shot 2019-09-26 at 15 44 28

Show the Google Analytics feature in the plans features lists
Screen Shot 2019-10-02 at 10 53 52

Update various nudges to reference the Premium plan, rather than Business plan, to enable Google Analytics
Screen Shot 2019-09-26 at 15 45 52

Screen Shot 2019-09-26 at 15 46 15

Adding the Google Analytics to the checkout thank you page

Screen Shot 2019-09-26 at 16 29 31

Adding Google Analytics to the Premium plan features
Screen Shot 2019-09-26 at 16 37 50

Note that this change updates the UI. D33243-code needs to be merged before Google Analytics will be functional for sites with a Premium plan.

##### Analytics settings

On a site with a Premium plan

- Visit `/marketing/traffic/{site}`
- See that the 3rd card on the page, "Google Analytics" allows setting a Tracking ID (in the format of GA-1234-1234)

##### Upgrade nudges

On a site with a Free or Personal plan

1. Visit `/marketing/traffic/{site}`
    - See that the 3rd card on the page, "Google Analytics" shows an upgrade nudge for the Premium plan
2. Visit `/stats/day/countryviews/{site}`
    - See that there's an upgrade nudge for the Premium plan

Repeat all of the above for a site with a Premium plan and see that no upgrade nudges are shown.

##### Plan features

On a site with a Free or Personal plan

- Visit `/plans/{site}` and upgrade to a Premium plan
- On the checkout thank you page (immediately after checkout), see Google Analytics listed as a Premium plan feature
- Click on "View all my features" at the top of the page to go to `/plans/my-plan/{site}`
- See Google Analytics listed as a Premium plan feature

Fixes #35737

### CODE REVIEW

File: client/my-sites/site-settings/form-analytics.jsx [hunk 1]
```diff
@@ -129,7 +117,7 @@ export class GoogleAnalyticsForm extends Component {
 			? translate(
 					'Connect your site to Google Analytics in seconds with Jetpack Premium or Professional'
 			  )
-			: translate( 'Connect your site to Google Analytics in seconds with the Business plan' );
+			: translate( 'Connect your site to Google Analytics in seconds with the Premium plan' );
```
Comment (a8ci18n) [inline client/my-sites/site-settings/form-analytics.jsx]: :ok: This change is safe, we'll keep the existing 17 translations.

**Comment by creativecoder** [comment]
I'm getting an issue after saving the Tracking ID and navigating to another page. I see a browser alert "Are you sure you want to leave this page?"--will take a look tomorrow.

**Comment by davemart-in** [comment]
I'll review this in the AM. Note: I also located a hand full of additional references 🙈

- https://en.blog.wordpress.com/2015/01/29/google-analytics-for-wordpress-com-business-sites/ (We might add an update to the top of this post)
- https://wordpress.com/go/digital-marketing/analyzing-organic-traffic-using-google-analytics/
- https://wordpress.com/google/analytics/ - Mentions "is included as part of WordPress.com Business"
- https://en.support.wordpress.com/google-analytics/ - "available as a feature of the WordPress.com Business plan" & "is part of the WordPress.com Business plan"
- https://en.support.wordpress.com/traffic/ - "upgrade to the WordPress.com Business plan" & this graphic: https://cl.ly/6dfd607fdba7
- https://en.support.wordpress.com/google-integration/ - "upgrading to the Business upgrade"
- https://wordpress.com/ - "design, Google Analytics, and" under business plans description
- https://wordpress.com/pricing/ - "design, Google Analytics, and" & "Google Analytics Integration" feature row
- https://wordpress.com/install-plugins/ - "concierge support, Google Analytics, and" under business plan description

@creativecoder I'm happy to help with these. Just let me know.

**Comment by davemart-in** [comment]
I know we also expose plans within the onboarding flow and after the new private by default site launch.

File: client/my-sites/stats/stats-module/index.jsx [hunk 2]
```diff
@@ -197,11 +197,10 @@ class StatsModule extends Component {
 					{ summary && 'countryviews' === path && (
 						<UpgradeNudge
 							title={ translate( 'Add Google Analytics' ) }
-							message={ translate(
-								'Upgrade to a Business Plan for Google Analytics integration.'
-							) }
+							message={ translate( 'Upgrade to a Premium Plan for Google Analytics integration.' ) }
```
Comment (pablinos) [inline client/my-sites/stats/stats-module/index.jsx]: I'm intrigued as to why this nudge is only on the countries view. Seems like we should put it on the others depending on how often people click it.

**Comment by pablinos** [comment]
> I'm getting an issue after saving the Tracking ID and navigating to another page. I see a browser alert "Are you sure you want to leave this page?"--will take a look tomorrow.

I was getting something like this, but it was because the save hadn't completed. Waiting and then trying again seemed to fix it. I wonder if we can wait for the save to complete before the navigation begins.

---

In general this is working well for me, and tests as expected. I think the other mentions of GA that @davemart-in has found are the marketing pages, so we can work through those separately.

**Comment by huguesvincent** [comment]
Some other pages where this is listed as well: 
1. https://wordpress.com/premium/
2. https://wordpress.com/create-blog/
3. https://wordpress.com/easy/
4. https://wordpress.com/create-website/
5. https://wordpress.com/built-for-business/
6. https://wordpress.com/business-everywhere/
7. https://wordpress.com/brand-everywhere/
8. https://wordpress.com/creativity-everywhere/
9. https://wordpress.com/everywhere/
10. https://wordpress.com/resolutions/

**Comment by scruffian** [comment]
> Last remaining bit will be to update the pricing tables in the onboarding flow and in the private by default upgrade flow.

Done in e0401b2a04

File: client/my-sites/site-settings/form-analytics.jsx (hunk 1 above)
Comment (scruffian) [reply client/my-sites/site-settings/form-analytics.jsx]: This seems worrying; shouldn't they be updated? I wonder if we can pull "Premium" out of the translation...

**Comment by creativecoder** [comment]
Note, I copied all of the urls outside of Calypso into these separate issues:

Marketing pages: https://github.com/Automattic/wp-calypso/issues/35740

Support docs: https://github.com/Automattic/wp-calypso/issues/35739

File: client/my-sites/site-settings/form-analytics.jsx (hunk 1 above)
Comment (creativecoder) [reply client/my-sites/site-settings/form-analytics.jsx]: I took that to mean that there were already translations in 17 languages for the updated string.

Comment (creativecoder) [reply client/my-sites/site-settings/form-analytics.jsx]: But I don't know, so I'm going to ask!

Comment (creativecoder) [reply client/my-sites/site-settings/form-analytics.jsx]: I found at that this means the English translation will be used until new translations are ready, so I've added the "String Freeze" label to this feature. We'll wait to deploy until new translations are ready.

**Pull request merged**